django/contrib/admin/static/admin/js/admin/RelatedObjectLookups.js

```diff
@@ -185,6 +187,56 @@
         win.close();
     }
 
+
+    function customEncodeURIComponent(s) {
```

Comment:
```suggestion
    function customEncodeURIComponent(s) {
        // Ensure that primary key values do not confuse the admin URLs by escaping problematic characters.
        // This is the JavaScript equivalent to django.contrib.admin.utils.quote().
```
We might need a similar reference in `django.contrib.admin.utils.quote` or the python `QUOTE_MAP` about this function (so people know if they update something there, they should also update here)